portality/models/account.py

```diff
@@ -133,3 +148,24 @@ def set_role(self, role):
 
     def prep(self):
         self.data['last_updated'] = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    @property
+    def api_key(self):
+        if self.has_role('api'):
+            return self.data.get('api_key', None)
+        else:
+            return None
+
+    def generate_api_key(self):
+        self.data['api_key'] = uuid.uuid4().hex
+
+    @classmethod
+    def pull_by_api_key(cls, key):
+        """Find a user by their API key - only succeed if they currently have API access."""
+        res = cls.query(q='api_key:"' + key + '"')
```

Comment:
.exact? Not that ES would try to analyse a 32 char continuous string, but still better to use the non-analysed field? (Since we thought that of usernames and then the space-at-the-end thing happened)